### DISCUSSION TITLE
Advantages of the timetravel feature

### DISCUSSION THREAD

**Discussion by hetzge** [cozodb/cozo#94] [General]
Timetravel is a cool and useful feature. But i feel a little bit like i miss something or do not fully understand the advantage. For me the timetravel feature is just syntactic sugar for an index over time. It is easy to understand but is it really worth it ? Couldn't i just model my data with `start` and `end` timestamp fields, index them and then query with `now() >= start AND now() < end` ? Ok in the case of the mood social network example from the [documentation](https://docs.cozodb.org/en/latest/releases/v0.4.html) it would be necessary to update the last entry end date when a new entry is added. So is this feature just to make this scenario easier or do i overlook something ? I am not sure if i really understand how special or not so special this feature is :)

**Comment by gisborne** [comment]
Storage is so cheap and the advantages so many that these days, most databases should be append-only with time travel, like this.

Even if we're just making it more convenient, this paradigm should be so central that making it easy is enough of a reason.

I'd like to see a generalisation of this in the longer run. So just as this feature restricts queries to a subset of the data across all participating relations, it would be great to be able to do this with a general predicate.

A great use case for this would be authorisation. I should be able to set a user token and then only see the data belonging to that user. There are other use cases.
